### ISSUE TITLE
Feature extraction

### ISSUE THREAD

**Issue by akashsharma29** [open-mmlab/mmaction#152]
I want to extract features for a custom dataset generated by TSN. I am trying to modify the 
mmaction/mmaction/models/recognizers/TSN2D.py file.

How can I remove the last FC layer, and obtain the features, (rather than the below numbers in the screenshot attached)?
![CustomDataUCFStyleAnnotnOutput](https://user-images.githubusercontent.com/36917739/78514540-8ab2ab00-7777-11ea-95da-3b995a5a4343.PNG)

**Comment by akashsharma29** [comment]
@yjxiong @acrophile @loveunk @lindahua Please reply to this. I could not find a similar resolved issue. Thanks!

**Comment by akashsharma29** [comment]
@kennymckormick Can you also look into the same and help me with the above. I just want features to be printed out. I want to use the above as a feature extractor.

**Comment by kennymckormick** [comment]
I think you should modify the forward_test code in TSN2D to generate feature instead of logits (not to forward fc). You also need to modify test_recognizer.py to collect and store the feature you extracted.

**Comment by mrlihellohorld** [comment]
> @kennymckormick Can you also look into the same and help me with the above. I just want features to be printed out. I want to use the above as a feature extractor.

could u tell me how u solved the problem（about extract feature）？
